src/main/java/core/basesyntax/Robot.java

```diff
@@ -1,4 +1,76 @@
 package core.basesyntax;
 
 public class Robot {
+    private Direction direction;
+    private int coordinateX;
+    private int coordinateY;
+
+    public Robot(Direction direction, int coordinateX, int coordinateY) {
+        this.direction = direction;
+        this.coordinateX = coordinateX;
+        this.coordinateY = coordinateY;
+    }
+
+    public Direction getDirection() {
+        return direction;
+    }
+
+    public int getCoordinateX() {
+        return coordinateX;
+    }
+
+    public int getCoordinateY() {
+        return coordinateY;
+    }
+
+    public void turnLeft() {
+        switch (direction) {
+            case UP:
+                direction = Direction.LEFT;
+                break;
+            case DOWN:
+                direction = Direction.RIGHT;
+                break;
+            case LEFT:
+                direction = Direction.DOWN;
+                break;
+            case RIGHT:
+                direction = Direction.UP;
+                break;
+        }
+    }
+
+    public void turnRight() {
+        switch (direction) {
+            case UP:
+                direction = Direction.RIGHT;
+                break;
+            case DOWN:
+                direction = Direction.LEFT;
+                break;
+            case LEFT:
+                direction = Direction.UP;
+                break;
+            case RIGHT:
+                direction = Direction.DOWN;
+                break;
+        }
+    }
+
+    public void stepForward() {
+        switch (direction) {
+            case UP:
+                coordinateX++;
```

Comment:
@EliosPride It seems you have mixed up x and y coordinates :)

Reply:
@al-pavlenko done.